### PR TITLE
ASAN and scan-build fixes

### PR DESCRIPTION
**Pull request by andyholmes**

### CODE REVIEW

File: libspiel/spiel-speaker.c [hunk 1]
```diff
@@ -1044,8 +1044,12 @@ _handle_gst_eos (GstBus *bus, GstMessage *msg, SpielSpeaker *self)
...
-  GstElement *fdsrc = entry->src;
-  gst_element_set_state (fdsrc, GST_STATE_NULL);
+
+  if (entry != NULL)
+    {
+      GstElement *fdsrc = entry->src;
+      gst_element_set_state (fdsrc, GST_STATE_NULL);
+    }
 
   return TRUE;
```
Comment (andyholmes) [inline libspiel/spiel-speaker.c:1054]: I believe this is correct, but the return value should maybe be `FALSE`. It seems to work as-is with some minor testing.

Comment (eeejay) [reply libspiel/spiel-speaker.c:1054]: I think true is correct, but I do think we need some kind of warning/assert if the entry is null.

File: libspiel/spiel-speaker.c [hunk 2]
```diff
@@ -1076,6 +1080,12 @@ _process_going_to_speak_message (GstMessage *msg, SpielSpeaker *self)
       g_warning ("No 'range_end' in message structure");
     }
 
+  if (entry == NULL)
+    {
+      g_debug ("Queue is empty");
+      return;
```
[the remark this answers is not in the document]
Comment (eeejay) [reply libspiel/spiel-speaker.c:1086]: Yeah, should happen earlier, and it should be a g_warning I believe.

**Review by eeejay — CHANGES_REQUESTED** [review]

**Comment by andyholmes** [comment]
I went with `g_return_if_fail()` variants for both, since it seems like a case for a critical-level message.

The message would be something like `assertion failed: entry != NULL`, but I think that's okay since this is pretty internal.

**Pull request merged**